From b60a050877d9483a44ffccf46aa430538a6d0aa9 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Sun, 30 Aug 2020 22:09:49 -0700
Subject: [PATCH 1547/1851] v4l: xilinx: vpss-scaler: Fix the compatible string
 deprecated message

commit 7ab44851ccc1f055d4c2b039495a0e95f20b8e3b from
https://github.com/Xilinx/linux-xlnx.git

The driver is currently printing the compatible string deprecated
message for all compatible strings incorrectly.
Instead do it for only "xlnx,v-vpss-scaler".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vpss-scaler.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
index 6a4ef88bea12..e278c04edf9d 100644
--- a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
@@ -1976,8 +1976,7 @@ static int xscaler_probe(struct platform_device *pdev)
 	if (!match)
 		return -ENODEV;
 
-	if (!strncmp(match->compatible, xscaler_of_id_table[0].compatible,
-		     strlen(xscaler_of_id_table[0].compatible))) {
+	if (!strcmp(xscaler_of_id_table[0].compatible, match->compatible)) {
 		dev_warn(&pdev->dev,
 			 "%s - compatible string is getting deprecated!\n",
 			 match->compatible);
-- 
2.31.1

